src/store/modules/layers.store.js

```diff
@@ -213,6 +213,42 @@ const getters = {
         const layer = state.activeLayers.find((layer) => layer.id === layerId)
         return layer?.isExternal || (layer?.type === LayerTypes.KML && !layer?.adminId)
     },
+
+    /**
+     * Returns true if the layer comes from a third party (external layer or KML layer) which has
+     * been imported from a local file.
+     *
+     * KML layer are treated as external when they are generated by another user (no adminId).
+     *
+     * @param {string} layerId Layer ID of the layer to check for data disclaimer
+     * @returns {Boolean}
+     */
+    hasDataDisclaimerLocal: (state) => (layerId) => {
+        const layer = state.activeLayers.find((layer) => layer.id === layerId)
+        const isBaseUrlValidUrl = /^\w+:\/\//.test(layer?.baseUrl)
+        return (
+            !isBaseUrlValidUrl &&
+            (layer?.isExternal || (layer?.type === LayerTypes.KML && !layer?.adminId))
+        )
+    },
+
+    /**
+     * Returns true if any layer comes from a third party (external layer or KML layer) which has
+     * been imported from a local file.
+     *
+     * KML layer are treated as external when they are generated by another user (no adminId).
+     *
+     * @returns {Boolean}
+     */
+    hasAnyExternalDataLocal: (state) => () => {
+        return state.activeLayers.some((layer) => {
+            const isBaseUrlValidUrl = /^\w+:\/\//.test(layer?.baseUrl)
+            return (
+                !isBaseUrlValidUrl &&
+                (layer.isExternal || (layer?.type === LayerTypes.KML && !layer?.adminId))
+            )
+        })
+    },
```

Comment:
I would also rename this one accordingly.
You should re-use the function above. You can get access to getters by adding them to the signature of the overall function
```Javascript
hasAnyLocaFile: (state, getters) => {
    return state.activeLayers.some((layer) => getters.isLocalFile(layer))
})
```